include/grpc/grpc_security.h

```diff
@@ -819,6 +819,7 @@ struct grpc_tls_credential_reload_arg {
   grpc_tls_key_materials_config* key_materials_config;
   grpc_ssl_certificate_config_reload_status status;
   const char* error_details;
+  grpc_tls_credential_reload_config* config;
```

Comment:
Few comments:
- can you add the new parameter "config" in the comment above to describe what it does?
- can you split the comment to describe each parameter on a new line?
- actual parameter name is key_materials_config but the comment says "key_materials" . Pls fix the discrepancy

Reply:
Another general comment: if this PR is for providing wrapper for C++ why are we making changes in the Core API that was part of a previous PR? Curious if the wrapper API needed this change

Reply:
The newest commit changes the comments, per your comment. 

On the general comment - the wrapper API needed this change to the Core API for the following reason: the user will provide C++ schedule/cancel functions to the constructors of `TlsCredentialReloadConfig` and `TlsServerAuthorizationCheckConfig` and, in order to pass these functions to the Core API, we require a mechanism for the C config to point to the C++ config that it arose from. This is where this `config` variable is used, as well as where the `context` variables added in `grpc_tls_credentials_options.h` are used. 

This is also explained in the design doc, which I will share with you.